src/shogun/lib/DynamicArray.h

```diff
@@ -405,9 +468,10 @@ template <class T> class CDynamicArray :public CSGObject
 		 *
 		 * @return the array
 		 */
-		inline T* get_array() const
+		inline T* get_array()
 		{
-			return m_array.get_array();
+			auto it = m_array.begin();
+			return &(*it);
```

Comment:
i'm quite sure that this is not correct, as elements in deque are not contiguous in memory.